pkg/api/types.go

```diff
@@ -1177,6 +1177,9 @@ type LoadBalancerIngress struct {
 
 // ServiceSpec describes the attributes that a user creates on a service
 type ServiceSpec struct {
+	// Type determines how the service will be exposed.  Valid options: ClusterIP, NodePort, LoadBalancer
+	Type ServiceType `json:"type,omitempty"`
+
```

Comment:
no, just what seems to make sense in terms of order of importance/requiredness..